From 4c02196e75b63030ce339a6ad7e4ff2314a10d84 Mon Sep 17 00:00:00 2001
From: Ricardo Neri <ricardo.neri@ti.com>
Date: Fri, 1 Jul 2011 11:59:06 -0500
Subject: [PATCH] ASoC: OMAP4: HDMI: Stop ACR WA before setting HDMI power state

The call to set hdmi_set_audio_power will cut the DSS clock if
HDMI video power state is OFF. Then, the ACR workaround must be
stopped before setting the HDMI power state to 0.

Signed-off-by: Ricardo Neri <ricardo.neri@ti.com>
Signed-off-by: Juan Gutierrez <jgutierrez@ti.com>
---
 sound/soc/omap/omap-hdmi.c |    9 ++++-----
 1 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/sound/soc/omap/omap-hdmi.c b/sound/soc/omap/omap-hdmi.c
index 5f7c50e..80ca9c0 100755
--- a/sound/soc/omap/omap-hdmi.c
+++ b/sound/soc/omap/omap-hdmi.c
@@ -145,14 +145,13 @@ static void omap_hdmi_dai_shutdown(struct snd_pcm_substream *substream,
 	hdmi_wrapper_disable(HDMI_WP);
 	hdmi_data.substream = NULL;
 
-	if (hdmi_data.active)
-		hdmi_set_audio_power(0);
-
-	#ifdef CONFIG_OMAP_HDMI_AUDIO_WA
+#ifdef CONFIG_OMAP_HDMI_AUDIO_WA
 	if (hdmi_lib_stop_acr_wa())
 		printk(KERN_WARNING "HDMI WA may be in bad state");
-	#endif
+#endif
 
+	if (hdmi_data.active)
+		hdmi_set_audio_power(0);
 
 #ifdef CONFIG_OMAP_PM
 	err = omap_pm_set_max_sdma_lat(&hdmi_data.qos_request,
-- 
1.7.1

